File: SimulationPython/Simulation/Abstract3DObject.py
import abc #for abstract classes
import numpy as np

class Abstract3DObject(metaclass=abc.ABCMeta):
   @abc.abstractmethod
   def __init__(self, x, y, z):
       self.pos = np.array([x, y, z], dtype=np.float64) #position of the 3D object in space

   def getPos(self):
       return self.pos

   def setPos(self, pos):
       self.pos = pos

File: SimulationCpp/SimulationCpp/src/Abstract3DObject.cpp
#include "Abstract3DObject.h"

Abstract3DObject::Abstract3DObject(double x, double y, double z)
	: m_pos{ Eigen::Vector3d{x, y, z} }
{
}

void Abstract3DObject::setPos(Eigen::Vector3d pos)
{
	m_pos = pos;
}